File: test-dest/29_smart_cast.swift

func main(args: [String]) {
  let x: Int32? = 1
  var y: Int32 = 2
  let z: Int32? = nil

  if let x = x {
    y += x
  } else {
    print("FAIL: x should not be nil")
  }

  if let z = z {
    print("FAIL: z should be nil")
    y += z
  }

  if y != 3 {
    print("FAIL: y should be 3, is \(y)")
  }
  print("Success")
}

main([])

File: test-src/29_smart_cast.kt
import java.util.*

fun main(args: Array<String>) {
  val x: Int? = 1
  var y: Int = 2
  val z: Int? = null

  if (x != null) {
    y += x
  } else {
    println("FAIL: x should not be null")
  }

  if (z != null) {
    println("FAIL: z should be null")
    y += z
  }

  if (y != 3) {
    println("FAIL: y should be 3, is $y")
  }
  println("Success")
}
